homeassistant/components/here_travel_time/__init__.py

```diff
@@ -137,14 +120,60 @@ def _update(self) -> HERERoutingData | None:
                     ATTR_DURATION_IN_TRAFFIC: traffic_time / 60,
                     ATTR_DISTANCE: distance,
                     ATTR_ROUTE: response.route_short,
-                    ATTR_ORIGIN: ",".join(here_formatted_origin),
-                    ATTR_DESTINATION: ",".join(here_formatted_destination),
+                    ATTR_ORIGIN: ",".join(origin),
+                    ATTR_DESTINATION: ",".join(destination),
                     ATTR_ORIGIN_NAME: waypoint[0]["mappedRoadName"],
                     ATTR_DESTINATION_NAME: waypoint[1]["mappedRoadName"],
                 }
             )
+        except InvalidCoordinatesException as ex:
+            _LOGGER.error("Could not call HERE api: %s", ex)
         return None
 
+    def _prepare_parameters(
+        self,
+    ) -> tuple[list[str], list[str], str | None, str | None]:
+        """Prepare parameters for the HERE api."""
+
+        if self.config.origin_entity_id is not None:
+            origin = find_coordinates(self.hass, self.config.origin_entity_id)
+        else:
+            origin = self.config.origin
+
+        if self.config.destination_entity_id is not None:
+            destination = find_coordinates(self.hass, self.config.destination_entity_id)
+        else:
+            destination = self.config.destination
+        try:
+            if destination is None:
+                raise vol.Invalid("Destination must not be None")
+            here_formatted_destination = destination.split(",")
+            vol.Schema(cv.gps(here_formatted_destination))
+        except (vol.Invalid) as ex:
+            raise InvalidCoordinatesException(
+                f"{destination} are not valid coordinates"
+            ) from ex
```

Comment:
Pull this out of the try block and raise InvalidCoordinatesException directly as I think the preference is to raise the exception directly even if it means duplicating some code.
```suggestion
        if destination is None:
            raise InvalidCoordinatesException("Destination is not configured")
        try:
            here_formatted_destination = destination.split(",")
            vol.Schema(cv.gps(here_formatted_destination))
        except (vol.Invalid) as ex:
            raise InvalidCoordinatesException(
                f"{destination} are not valid coordinates"
            ) from ex
```

Reply:
Done.
The function looks a bit complex now. I have several PRs lined up that will deal with that.